cdap-api/src/main/java/io/cdap/cdap/api/security/credential/ProvisionedCredential.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.api.security.credential;
+
+/**
+ * A credential provisioned by the {@link CredentialProvider}.
+ */
+public class ProvisionedCredential {
+
+  private final String value;
+  private final long expirationSecs;
+
+  /**
+   * Creates a provisioned credential.
+   *
+   * @param value          The credential value.
+   * @param expirationSecs The expiration timestamp of the credential in seconds.
+   */
+  public ProvisionedCredential(String value, long expirationSecs) {
+    this.value = value;
+    this.expirationSecs = expirationSecs;
+  }
+
+  /**
+   * Returns the credential value.
+   *
+   * @return The credential value.
+   */
+  public String get() {
+    return value;
+  }
+
+  /**
+   * Returns the credential expiration timestamp in seconds.
+   *
+   * @return The credential expiration timestamp in seconds.
+   */
+  public long getExpirationSecs() {
```

Comment:
can we use an `Instant` instead? Then it removes the ambiguity

Reply:
Changed the type to `Instant` and renamed to `expiration` to avoid ambiguity.